joomla/joomla-cms
# [4.0] [a11y] mod_menu aria-current page/location

Pull Request for Issue # .

### Summary of Changes

With https://github.com/joomla/joomla-cms/pull/29343 I've implemented `aria-current="page"`
But... the menu item is not necessary the current page.  => see https://github.com/joomla/joomla-cms/pull/29343#issuecomment-636795766

Therefor I have rewritten the `aria-current="page"` implementation by adding some logic to the `foreach` loop of items in the `getList` function of MenuHelper. Added comparing the url parameters `option`, `view` and `id` (if applicable) with those from the menu item.

If the comparison results true, the new field `$item->current` changes from `false` into `true`

### Testing Instructions

- Joomla 4.0 beta with sample data installed (forgive me ... I don't know which sample data, since I've pressed all of them.)
- Click on menu item `blog`
- Open your browser develop tools and find the output of mod_menu
- Find the anchor of menu item `blog`, the current page and see the attribute `aria-current="page"` 

- Click on article `Welcome to your blog`
- Open your browser develop tools and find the output of mod_menu
- Find the anchor of menu item `blog`, the current page and see the attribute `aria-current="page"` 

- A menu item to the login page has no `id`. The solution handles a non-set item->id.
- The `$item->current` is not set to true if the menu item is not connected to the output of a component... for instance an alias item.

### Expected result

Expected is that menu item `blog` does not have attribute `aria-current="page"` when visiting the blog article `Welcome to our blog`.

### Actual result

Both the menu item of the category blog as the articles of this menu item contain attribute `aria-current="blog"`.

After applying the patch, the actual result is equal to the expected result. 

### Documentation Changes Required

no documentation changes required.
